apps/consumer-client/src/pages/pcdpass-examples/zk-eddsa-event-ticket-proof.tsx

```diff
@@ -0,0 +1,468 @@
+      value: undefined,
+      userProvided: true
+    },
+    validEventIds: {
+      argumentType: ArgumentTypeName.StringArray,
+      value: validEventIds.length != 0 ? validEventIds : undefined,
+      userProvided: false
+    },
+    fieldsToReveal: {
+      argumentType: ArgumentTypeName.Object,
+      value: fieldsToReveal,
+      userProvided: false
+    },
+    watermark: {
+      argumentType: ArgumentTypeName.BigInt,
+      value: watermark.toString(),
+      userProvided: false
+    }
+  };
+
+  if (externalNullifier) {
+    args.externalNullifier = {
+      argumentType: ArgumentTypeName.BigInt,
+      value: externalNullifier,
+      userProvided: false
+    };
+  }
+
+  const proofUrl = constructPassportPcdGetRequestUrl<
+    typeof ZKEdDSAEventTicketPCDPackage
+  >(urlToPassportWebsite, popupUrl, ZKEdDSAEventTicketPCDPackage.name, args, {
+    genericProveScreen: true,
+    title: "ZKEdDSA Proof",
+    description: "zkeddsa ticket pcd request"
+  });
+
+  openPassportPopup(popupUrl, proofUrl);
+}
+
+/**
+ * React hook which can be used on 3rd party application websites that
+ * parses and verifies a PCD representing a ZKEdDSA ticket proof.
+ */
+function useZKEdDSAEventTicketProof(
+  pcdStr: string,
+  onVerified: (valid: boolean) => void,
+  fieldsToReveal: EdDSATicketFieldsToReveal,
+  watermark: bigint,
+  externalNullifier?: string
+): { pcd: ZKEdDSAEventTicketPCD | undefined; error: any } {
+  const [error, _setError] = useState<Error | undefined>();
+  const zkEdDSAEventTicketPCD = useSerializedPCD(
+    ZKEdDSAEventTicketPCDPackage,
+    pcdStr
+  );
+
+  useEffect(() => {
+    if (zkEdDSAEventTicketPCD) {
+      verifyProof(
+        zkEdDSAEventTicketPCD,
+        fieldsToReveal,
+        watermark,
+        externalNullifier
+      ).then(onVerified);
+    }
+  }, [
+    zkEdDSAEventTicketPCD,
+    fieldsToReveal,
+    watermark,
+    externalNullifier,
+    onVerified
+  ]);
+
+  return {
+    pcd: zkEdDSAEventTicketPCD,
+    error
+  };
+}
+
+async function verifyProof(
+  pcd: ZKEdDSAEventTicketPCD,
+  fieldsToReveal: EdDSATicketFieldsToReveal,
+  watermark: bigint,
+  externalNullifier?: string
+): Promise<boolean> {
+  // TODO: after eddsa keys are represented in normal form and not montgomery
+  // form, we can remove the call to Package initialization logic and also
+  // remove the package artifacts in the ../../public directory (for now)
+
+  // ordinarily, we wouldn't need to init the package to call verify
+  // however, because converting the claim into public signals for verification
+  // requires us to convert the signing key from montgomery form, we need
+  // to use methods on circomlibjs.babyJub.F. and to do this, we need to
+  // build babyJub, which happens in init
+
+  // once we've made it so that keys are no longer in montgomery form, we won't
+  // need to do this anymore
+  // in general i am of the opinion that verification shouldn't require you
+  // to call init...
```

Comment:
I understand this perspective. However, do you think it makes sense for each `PCDPackage` to have its own semantics for when and if `init` should be called? I think that may get confusing. Eg. what if some package absolutely positively *does* need `init` called before using `verify`? The reason you don't strictly need to here is an implementation detail of `snarkjs`.

Reply:
this is a copy of my comment from the old zkeddsa pcd. not sure if @artwyman shares this opinion or just copied the comment

Reply:
Now that I understand this better, I figured out how to restructure the code to remove the unnecessary dependency now.  Will include in a follow-up PR.